### PR TITLE
Modifications to PHI Estimator 

### PR DESCRIPTION
**Pull request by ashishsingh18**
Modifications to PHI Estimator 
-input selection via combo
-data checks
-results in UI

### CODE REVIEW

File: Plugins/org.mitk.cbica.captk.phiestimator/src/internal/QmitkPHIEstimatorView.cpp [hunk 1]
```diff
@@ -67,16 +73,92 @@ namespace
...
+}
+
+QmitkPHIEstimatorView::~QmitkPHIEstimatorView()
+{
+
+}
+
 void QmitkPHIEstimatorView::CreateQtPartControl(QWidget* parent)
 {
   // Setting up the UI is a true pleasure when using .ui files, isn't it?
```
Comment (dboun) [inline Plugins/org.mitk.cbica.captk.phiestimator/src/internal/QmitkPHIEstimatorView.cpp:119]: 🤣🤣🤣🤣

File: Plugins/org.mitk.cbica.captk.phiestimator/src/internal/QmitkPHIEstimatorView.cpp [hunk 2]
```diff
@@ -91,13 +173,22 @@ void QmitkPHIEstimatorView::OnSelectionChanged(berry::IWorkbenchPart::Pointer, c
...
-  m_Controls.selectImageLabel->setVisible(true);
+}
+
+void QmitkPHIEstimatorView::OnReferenceSelectionChanged(const mitk::DataNode * node)
+{
+	m_Controls.groupBox_Result->hide();
+	this->ResetResults();
+}
+
+void QmitkPHIEstimatorView::OnSegmentationSelectionChanged(const mitk::DataNode * node)
```
Comment (dboun) [inline Plugins/org.mitk.cbica.captk.phiestimator/src/internal/QmitkPHIEstimatorView.cpp:188]: You should probably add the checks for if there are two labels here

File: Plugins/org.mitk.cbica.captk.phiestimator/src/internal/QmitkPHIEstimatorView.cpp [hunk 3]
```diff
@@ -153,40 +258,11 @@ if (maskdata == nullptr)
...
-      auto imageName = firstSelectedDataNode->GetName();
-
-      MITK_INFO << "Process image \"" << imageName << "\" ...";
-
       // We're finally using the ExampleImageFilter
  
 	  //typename PerfusionImageType::Pointer pimg = PerfusionImageType::New();
 	  //mitk::CastToItkImage(image, pimg);
-	  PerfusionImageType::Pointer pimg = mitk::ImageToItkImage<short,4>(image);
+	  PerfusionImageType::Pointer pimg = mitk::ImageToItkImage<short,4>(rimage);
```
Comment (dboun) [inline Plugins/org.mitk.cbica.captk.phiestimator/src/internal/QmitkPHIEstimatorView.cpp:265]: try / catch with:

```cpp
  catch (const mitk::Exception& e)
  {
    MITK_ERROR << "MITK Exception: " << e.what();
  }
  catch (...)
  {
    std::cerr << "Unexpected error!";
    return EXIT_FAILURE;
  }
```

Comment (dboun) [reply Plugins/org.mitk.cbica.captk.phiestimator/src/internal/QmitkPHIEstimatorView.cpp:265]: include those:

#include <mitkException.h>
#include <mitkLogMacros.h>

Comment (dboun) [reply Plugins/org.mitk.cbica.captk.phiestimator/src/internal/QmitkPHIEstimatorView.cpp:265]: also check if 4D actually exists

Comment (dboun) [reply Plugins/org.mitk.cbica.captk.phiestimator/src/internal/QmitkPHIEstimatorView.cpp:265]: also check if seg is 3D with similar params

File: Plugins/org.mitk.cbica.captk.phiestimator/src/internal/QmitkPHIEstimatorView.cpp (hunk 2 above)
Comment (ashishsingh18) [reply Plugins/org.mitk.cbica.captk.phiestimator/src/internal/QmitkPHIEstimatorView.cpp:188]: I was planning to add these on the run button callback.

File: Plugins/org.mitk.cbica.captk.phiestimator/src/internal/QmitkPHIEstimatorView.cpp (hunk 3 above)
Comment (ashishsingh18) [reply Plugins/org.mitk.cbica.captk.phiestimator/src/internal/QmitkPHIEstimatorView.cpp:265]: Good catch..we need the exceptions handled and logged as well

Comment (ashishsingh18) [reply Plugins/org.mitk.cbica.captk.phiestimator/src/internal/QmitkPHIEstimatorView.cpp:265]: > 
> 
> also check if 4D actually exists

There is a separate issue open for this. #71

Comment (ashishsingh18) [reply Plugins/org.mitk.cbica.captk.phiestimator/src/internal/QmitkPHIEstimatorView.cpp:265]: > 
> 
> also check if seg is 3D with similar params

There is a separate issue open for this, #72

Comment (ashishsingh18) [reply Plugins/org.mitk.cbica.captk.phiestimator/src/internal/QmitkPHIEstimatorView.cpp:265]: The other exception handling may not be needed once we have #71, #72. But in general it is good to have it.

**Review by dboun — APPROVED** [review]

**Pull request merged**